core/src/main/java/org/apache/carbondata/core/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -586,16 +586,27 @@ private int getKeySize(List<QueryDimension> queryDimension,
    */
   @Override public void finish() throws QueryExecutionException {
     CarbonUtil.clearBlockCache(queryProperties.dataBlocks);
+    UnsafeMemoryManager.INSTANCE.freeMemoryAll(ThreadLocalTaskInfo.getCarbonTaskInfo().getTaskId());
```

Comment:
Better move down it to after closing of queryIterator

Reply:
ok